src/codewhisperer/service/securityIssueHoverProvider.ts

```diff
@@ -41,6 +41,44 @@ export class SecurityIssueHoverProvider implements vscode.HoverProvider {
         return new vscode.Hover(contents)
     }
 
+    /**
+     * Updates the position of each hover when the text document is changed. The underlying security issues
+     * will be updated with new line numbers to reflect the new positions of the hover.
+     *
+     * @param event Event that triggered the text document change
+     */
+    public updateRanges(event: vscode.TextDocumentChangeEvent) {
+        const changedRange = event.contentChanges[0].range
+        const changedText = event.contentChanges[0].text
+        const lineOffset = this._getLineOffset(changedRange, changedText)
+
+        this._issues = this._issues.map(issues => this._applyRangeOffset(event.document.fileName, issues, lineOffset))
+    }
+
+    private _getLineOffset(range: vscode.Range, text: string) {
+        const originLines = range.end.line - range.start.line + 1
+        const changedLines = text.split('\n').length
+        return changedLines - originLines
+    }
+
+    private _applyRangeOffset(
+        fileName: string,
+        aggregatedIssues: AggregatedCodeScanIssue,
+        lineOffset: number
+    ): AggregatedCodeScanIssue {
+        if (aggregatedIssues.filePath !== fileName) {
+            return aggregatedIssues
+        }
+        return {
+            ...aggregatedIssues,
+            issues: aggregatedIssues.issues.map(issue => ({
+                ...issue,
+                startLine: issue.startLine + lineOffset,
```

Comment:
the line numbers should only be updated if the text change in file happened before this startLine. Seems like this will update the start and end lines for some issues which are not impacted by the text change in the file.

e.g., if change of text in a file is between line numbers 25 and 30 then the issues should only be updated starting from line number 25.

Reply:
Good catch, updating it